src/west/app/project.py

```diff
@@ -867,6 +867,16 @@ def fetch_missing_imports(self, args):
         self.manifest = Manifest.from_file(topdir=self.topdir,
                                            importer=self.update_importer)
 
+    def update_submodules(self, project, submodules, call_location):
+        # Updates given list of project submodules by using
+        # 'git submodule update --init' command from the location passed
+        # through the call_location parameter value.
+        if submodules:
+            # Iterate through the submodules list and update each one of them
+            for submodule in submodules:
```

Comment:
we are sure that always type of submodules variable is list ?

Reply:
I think so, as it is sequence type and I checked that it is translated in python to the list. I was basing on the 'projects' solution having sequence type and in other west methods implementations we are directly iterating through the list without any previous checks. If you find it insecure we should fix it in many other places too.

Reply:
:+1: